src/logic/LogicGate.tsx

```diff
@@ -41,7 +41,24 @@ function isNot(type: GateType) {
     case GateType.BUF:
       return false;
     default:
-      throw Error("Unsupported Gate Type")
+      throw new Error(`Unsupported Gate Type(${type})`)
+  }
+}
+
+function maxPins(type: GateType): number {
```

Comment:
how will this work if we have dynamic resizing (width, etc)?

Reply:
We'll filter in the UI when these properties are changed, and we may also filter when these values are changed.  For now, width and fieldWidth are accessible and modifiable as properties.